Clean/Utils/Tactics/DecomposeProvableStruct.lean

```diff
@@ -0,0 +1,204 @@
+      | _ => return false
+
+    -- Search for projections in the types of hypotheses
+    for localDecl in ctx do
+      if localDecl.isImplementationDetail then
+        continue
+
+      -- Search in the type of each hypothesis
+      let varsInType ← findStructVars localDecl.type
+      -- Filter to only include those with ProvableStruct instances
+      for fvarId in varsInType do
+        if ← hasProvableStructInstance fvarId then
+          result := result ++ [fvarId]
+
+    -- Also search for projections in the goal
+    let goal ← getMainGoal
+    let target ← goal.getType
+    let varsInGoal ← findStructVars target
+    -- Filter to only include those with ProvableStruct instances
+    for fvarId in varsInGoal do
+      if ← hasProvableStructInstance fvarId then
+        result := result ++ [fvarId]
+
+    -- Remove duplicates and return
+    return result.eraseDups
+
+/--
+  Decompose all variables with ProvableStruct instances in the context
+-/
+def decomposeProvableStruct : Lean.Elab.Tactic.TacticM Unit := do
+  withMainContext do
+    let goal ← getMainGoal
+
+    -- Find all variables with ProvableStruct instances that appear in projections
+    let fvarIds ← findProvableStructVars
+
+    if fvarIds.isEmpty then
+      throwError "No variables with ProvableStruct instances found in the context or goal"
+
+    -- Apply cases on each variable
+    let mut currentGoal := goal
+    let mut decomposed := false
+
+    for fvarId in fvarIds do
+      let localDecl ← fvarId.getDecl
+      let userName := localDecl.userName
+
+      try
+        -- Use cases tactic on the variable
+        let casesResult ← currentGoal.cases fvarId
+
+        -- Get the new goal from the cases result
+        match casesResult.toList with
+        | [subgoal] =>
+          currentGoal := subgoal.mvarId
+          decomposed := true
+        | _ =>
+          throwError s!"Unexpected result from cases on {userName}"
+      catch e =>
+        -- Skip variables that can't be decomposed
+        -- This can happen with type synonyms or other non-destructurable types
+        trace[Meta.Tactic] "Failed to decompose struct variable {userName}: {e.toMessageData}"
+        continue
+
+    if not decomposed then
+      throwError "Failed to decompose any variables with ProvableStruct instances"
+
+    replaceMainGoal [currentGoal]
+
+
+/--
+  Automatically decompose ALL variables with ProvableStruct instances that either:
+  1. Are in the context as variables with ProvableStruct instances
+  2. Appear in field projections in hypotheses (e.g., h : input.x = 5)
+  3. Appear in field projections in the goal
+
+  Note: 
+  - The new variables introduced by decomposition are not explicitly named. 
+    Use `rename_i` after this tactic to give them names.
+  - This tactic only performs one level of decomposition. For nested structures,
+    it will decompose the outer structure but not the inner ones. Use under
+    `repeat` to decompose nested structures fully.
```

Comment:
```suggestion
  - This tactic only performs one level of decomposition. For nested structures,
    it will decompose the outer structure but not the inner ones. Use under
    `repeat` to decompose nested structures fully. `provable_struct_simp` is one
    implementation of such a loop.
```